Sources/StreamVideo/Utils/AudioSession/AudioDeviceModule/AudioEngineLevelNodeAdapter.swift

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright © 2025 Stream.io Inc. All rights reserved.
+//
+
+import Accelerate
+import AVFoundation
+import Combine
+import Foundation
+
+protocol AudioEngineNodeAdapting {
+
+    var subject: CurrentValueSubject<Float, Never>? { get set }
+
+    func installInputTap(
+        on node: AVAudioNode,
+        format: AVAudioFormat,
+        bus: Int,
+        bufferSize: UInt32
+    )
+
+    func uninstall(on bus: Int)
+}
+
+/// Observes an `AVAudioMixerNode` and publishes decibel readings for UI and
+/// analytics consumers.
+final class AudioEngineLevelNodeAdapter: AudioEngineNodeAdapting {
+
+    var subject: CurrentValueSubject<Float, Never>?
+
+//    private let publisher: (Float) -> Void
+    private var inputTap: AVAudioMixerNode?
+
+    /// Installs a tap on the supplied audio node to monitor input levels.
+    /// - Parameters:
+    ///   - node: The node to observe; must be an `AVAudioMixerNode`.
+    ///   - format: Audio format expected by the tap.
+    ///   - bus: Output bus to observe.
+    ///   - bufferSize: Tap buffer size.
+    func installInputTap(
+        on node: AVAudioNode,
+        format: AVAudioFormat,
+        bus: Int = 0,
+        bufferSize: UInt32 = 1024
+    ) {
+        guard let mixer = node as? AVAudioMixerNode, inputTap == nil else { return }
+
+        mixer.installTap(
+            onBus: bus,
+            bufferSize: bufferSize,
+            format: format
+        ) { [weak self] buffer, _ in
+            self?.processInputBuffer(buffer)
+        }
+
+        inputTap = mixer
+        log.debug("Input node installed", subsystems: .audioRecording)
+    }
+
+    /// Removes the tap and resets observed audio levels.
+    /// - Parameter bus: Bus to remove the tap from, defaults to `0`.
+    func uninstall(on bus: Int = 0) {
+        if let mixer = inputTap, mixer.engine != nil {
+            mixer.removeTap(onBus: 0)
+        }
+        subject?.send(-160)
```

Comment:
this is silence, right? Similarly to the other one, best to have constants for readability.